rezerwacje-app/src/main/java/org/niewidoczniakademicy/rezerwacje/service/RecruitmentService.java

```diff
@@ -0,0 +1,97 @@
+package org.niewidoczniakademicy.rezerwacje.service;
+
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.niewidoczniakademicy.rezerwacje.model.database.Recruitment;
+import org.niewidoczniakademicy.rezerwacje.model.database.RecruitmentPeriod;
+import org.niewidoczniakademicy.rezerwacje.model.database.Room;
+import org.niewidoczniakademicy.rezerwacje.model.rest.other.RecruitmentAndRecruitmentPeriodConnectionResponse;
+import org.niewidoczniakademicy.rezerwacje.model.rest.other.RecruitmentAndRoomConnectionResponse;
+import org.niewidoczniakademicy.rezerwacje.model.rest.recruitment.AddRecruitmentRequest;
+import org.niewidoczniakademicy.rezerwacje.model.rest.recruitment.AddRecruitmentResponse;
+import org.niewidoczniakademicy.rezerwacje.model.rest.recruitment.GetRecruitmentResponse;
+import org.niewidoczniakademicy.rezerwacje.service.converter.ConversionService;
+import org.niewidoczniakademicy.rezerwacje.service.exception.RecruitmentNotFoundException;
+import org.niewidoczniakademicy.rezerwacje.service.exception.StartDateNotBeforeEndDateException;
+import org.niewidoczniakademicy.rezerwacje.service.repository.RecruitmentPeriodRepository;
+import org.niewidoczniakademicy.rezerwacje.service.repository.RecruitmentRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Slf4j
+@Service
+@AllArgsConstructor(onConstructor = @__(@Autowired))
+public final class RecruitmentService {
+
+    private final RecruitmentRepository recruitmentRepository;
+    private final RecruitmentPeriodRepository recruitmentPeriodRepository;
+
+    private final ConversionService conversionService;
+    private final RoomService roomService;
+    private final RecruitmentPeriodService recruitmentPeriodService;
+
+    public AddRecruitmentResponse saveRecruitment(final AddRecruitmentRequest request) {
+        validateAddRecruitmentRequest(request);
+
+        final Recruitment recruitment = conversionService.convert(request);
+        recruitmentRepository.save(recruitment);
+
+        return AddRecruitmentResponse.builder()
+                .recruitmentId(recruitment.getId())
+                .build();
+    }
+
+    public GetRecruitmentResponse getRecruitmentByName(final String name) {
+        final Recruitment recruitment = getRecruitmentFromDatabaseByName(name);
+
+        return GetRecruitmentResponse.builder()
+                .recruitment(recruitment)
+                .build();
+    }
+
+    public RecruitmentAndRoomConnectionResponse connectRecruitmentAndRoom(final String name,
```

Comment:
Shouldn't we use Id for the recruitment as well? String may be ambiguous

Reply:
Done - corrected also connection between user and course of study